src/xSTIR/pSTIR/STIR.py

```diff
@@ -1321,6 +1346,16 @@ def modality(self):
         """Returns imaging modality as Python string."""
         return parms.char_par(self.handle, 'AcquisitionData', 'modality')
 
+    @property
+    def __array_interface__(self):
+        """As per https://numpy.org/doc/stable/reference/arrays.interface.html"""
+        return {'shape': self.shape, 'typestr': '<f4', 'version': 3,
+                'data': (parms.size_t_par(self.handle, 'AcquisitionData', 'address'), False)}
+
+    def asarray(self, xp=numpy, **kwargs):
+        """Returns view of self"""
+        return xp.asarray(self, **kwargs)
+
     def as_array(self):
         """Returns bin values as ndarray.
```

Comment:
probably should say this makes a new `numpy.ndarray` and recommend using `asarray` instead (with `copy=True` if required)

Reply:
what about something like:

```py
def as_array(self, warn=True):
    if warn:
        warnings.warn(UserWarning, "use `asarray` instead")
    ...
```

Reply:
if we do that, then I guess we'd need
```
warnings.warn("deprecated: use `asarray` instead", DeprecationWarning)
```
and tell users they can filter the warning using
```
import warnings

warnings.filterwarnings('ignore', category=DeprecationWarning)
```
However, that'll only work if we never use `as_array()` ourselves anymore. I therefore suggest we first do this via docstring and CHANGES.md warning now, and really deprecate in the major release afterwards.

Reply:
Hence the `warn=True` kwarg which we can set to `False` to suppress warnings.

Alternatively we can `with warnings.catch_warnings()` contextmanager to temporarily suppress.

Reply:
```suggestion
        """
        WARNING: you probably should use `.asarray()` (no underscore) instead.

        Returns bin values as ndarray.
```